magda-content-api/src/createApiRouter.ts

```diff
@@ -35,14 +45,52 @@ export default function createApiRouter(options: ApiRouterOptions) {
      * @apiSuccessExample {json} 200
      *    [
      *        {
-     *            "id": ...,
+     *            "id": ...
      *            "type": ...
+     *            "length": ...
      *        },
      *        ...
      *    ]
      */
-    router.get("/all", ADMIN, async function(req, res) {
-        res.json(await database.getContentSummary());
+    router.get("/all", USER, async function(req, res) {
+        // get summary
+        let all = await database.getContentSummary();
```

Comment:
Can we do this filtering with an SQL query, or does it require something that the wildcard library does that SQL pattern matching doesn't do?

Then we could just hit the records we need, and do the inlining in the same trip.

Reply:
Can I please keep it 🙏 ? My rationale:

- Right now DB is just a key value store.
- Want to move DB to an even sourcing model with caching all content in memory. This works perfectly for that.